src/libopenrave/kinbody.cpp

```diff
@@ -797,7 +797,10 @@ bool KinBody::InitFromKinBodyInfo(const KinBodyInfo& info)
         return false;
     }
 
-    _id = info._id;
+    if( !info._id.empty() ) {
```

Comment:
@rdiankov Unfortunately, `info` here is `const`, so we cannot simply update its `id` before passing in.